src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -250,6 +255,44 @@ export class KycService {
     return KycStepMapper.toKycResult(kycStep);
   }
 
+  async updateNationalityData(kycHash: string, stepId: number, data: KycNationalityData): Promise<KycResultDto> {
+    let user = await this.getUser(kycHash);
+    const kycStep = user.getPendingStepOrThrow(stepId);
+
+    const updatedUser = await this.userDataService.updateNationality(user, data);
+    user = updatedUser.nationality ? updatedUser.failStep(kycStep, data) : updatedUser.completeStep(kycStep, data);
```

Comment:
`updateNationality` throws an error, if the country is not found, so this check looks useless 🤔